plugin/pkg/admission/initialization/initialization.go

```diff
@@ -43,38 +54,131 @@ type initializerOptions struct {
 	Initializers []string
 }
 
+type InitializationConfig interface {
+	Run(stopCh <-chan struct{})
+	Initializers() (*v1alpha1.InitializerConfiguration, error)
+}
+
 type initializer struct {
-	resources  map[schema.GroupResource]initializerOptions
+	config     InitializationConfig
 	authorizer authorizer.Authorizer
 }
 
-// NewAlwaysAdmit creates a new always admit admission handler
+// Retry config loading failures for up to four and a half seconds if the config hasn't been loaded
+// yet or if the server is down. Creation requests are delayed during this interval, which prevents
+// racy failures during startup until the initializer configuration becomes available.
+// TODO: move into InitializationConfigurationManager, since these values depend on the config
+//   refresh loop.
+const (
+	retryTemporaryConfigFailures = 8
```

Comment:
I see. In that case, can we let the CM count how long it has been running and return a specific "not initialized" error if it's been up for less than 4.5s, and here we only run the retry loop if CM returns the "not initialized" error? The current implementation retries even after the startup phase.

I'm thinking if the same logic is needed by the genearic admission controller. I guess it's not necessary, so i'm fine with the TODO.